src/Matroids/matroids.jl

```diff
@@ -0,0 +1,322 @@
+	#	throw("Input is not a collection of circuits")
+	#end
+    	return Matroid(M,groundset,gs2num)
+end
+
+@doc Markdown.doc"""
+A matroid represented by the column vectors of a matrix.
+
+# Examples
+To construct the linear matroid of the matrix `A` over the field with two elements write:
+```jldoctest
+julia> A = [[1,0],[0,1],[1,1],[1,1]]
+
+julia> M = matroid_from_matrix(matrix(GF(2),A))
+```
+
+or to obtain the same result
+```jldoctest
+julia> A = [[1,1],[1,1]]
+
+julia> M = matroid_from_reduced_matrix(matrix(GF(2),A))
+```
+"""
+matroid_from_reduced_matrix(A::MatrixElem) = matroid_from_matrix(hcat(identity_matrix(A.base_ring,nrows(A)),A))
+
+function matroid_from_matrix(A::MatrixElem)
+	rk = rank(A)
+	bases = Vector{Vector{Int}}()
+	for set in Oscar.Hecke.subsets(Vector(1:ncols(A)),rk)
+		if rank(A[:,set])==rk
+			push!(bases, set);
+		end
+	end
+	return matroid_from_bases(bases,ncols(A))
+end
+
+@doc Markdown.doc"""
+The cycle matroid of a graph.
+
+# Examples
+To construct the cycle matroid of the complete graph of 4 vertices write:
+```jldoctest
+julia> g = Oscar.Graphs.complete_graph(4)
+
+julia> M = cycle_matroid(g)
+```
+"""
+function cycle_matroid(g::Oscar.Graphs.Graph)
+	pm_Graph = Polymake.graph.Graph(ADJACENCY=g.pm_graph)
+	M = Polymake.matroid.matroid_from_graph(pm_Graph)
+	n = Oscar.Graphs.ne(g)
+	gs2num = Dict{Any,Int}()
+	i = 1
+	for elem in 1:n
+		gs2num[elem] = i
+		i+=1
+	end
+	return Matroid(M,1:n,gs2num)
+end
+
+@doc Markdown.doc"""
+The bond or cocylce matroid of a graph which is the dual of a cycle matroid, e.g, cographic.
+
+# Examples
+To construct the bond or cocycle matroid of the complete graph of 4 vertices write:
+```jldoctest
+julia> g = Oscar.Graphs.complete_graph(4)
+
+julia> M = bond_matroid(g)
+```
+
+or
+```jldoctest
+julia> M = cocycle_matroid(g)
+```
+"""
+bond_matroid(g::Oscar.Graphs.Graph) = dual_matroid(cycle_matroid(g))
+cocycle_matroid(g::Oscar.Graphs.Graph) = bond_matroid(g::Oscar.Graphs.Graph)
+
+
+@doc Markdown.doc"""
+The dual matroid of a given matroid.
+
+# Examples
+To construct the dual of the fano matroid write:
+```jldoctest
+julia> M = dual_matroid(fano_matroid())
+```
+"""
+dual_matroid(M::Matroid) = Matroid(M.pm_matroid.DUAL,M.groundset,M.gs2num)
+
+@doc Markdown.doc"""
+The ground set of a matroid.
+
+To obtain the ground set of the fano matroid type:
+# Example
+```jldoctest
+julia> ground_set(fano())
+```
+"""
+groundset(M::Matroid) = M.groundset
+
+@doc Markdown.doc"""
+The direct sum of matroids.
+
+To obtain the direct sum of the fano and non-fano matroid type:
+# Example
+```jldoctest
+julia> direct_sum(fano_matroid(),non_fano_matroid())
+```
+
+or to take the sum of more matroids use:
+# Example
+```jldoctest
+julia> matroids = Vector(uniform_matroid(2,4), uniform_matroid(1,3), uniform_matroid(3,4))
+
+julia> direct_sum(matroids)
+```
+"""
+function direct_sum(M::Matroid, N::Matroid)
+	gsN = N.groundset
+	while(size( intersect(M.groundset,gsN) )[1]>0)
```

Comment:
Aren't these `Vector` objects? If so, I'd use `length`, or `isempty`; for that matter, since you don't need to compoute the full intersection, perhaps
```suggestion
	while any(in(M.groundset),gsN)
```